File: android/app/src/main/kotlin/com/example/flutter_interactive/MyBasicChannel.kt
package com.example.flutter_interactive

import android.util.Log
import io.flutter.plugin.common.BasicMessageChannel
import io.flutter.plugin.common.StringCodec
import io.flutter.plugin.common.BinaryMessenger

class MyBasicChannel(messenger: BinaryMessenger) : BasicMessageChannel.MessageHandler<String> {

    private var myChannel: BasicMessageChannel<String> =
            BasicMessageChannel(messenger, "com.flutter.test.basic_channel", StringCodec.INSTANCE)

    init {
        myChannel.setMessageHandler(this)
    }


    override fun onMessage(message: String?, reply: BasicMessageChannel.Reply<String>) {
        Log.e("android", message!!)

        myChannel.send("from android Message", reply)
    }
}

File: ios/Runner/MyBasicChannel.swift
//
//  MyBasicChannel.swift
//  Runner
//
//  Created by 福建连 on 7/28/21.
//

import Foundation
import UIKit

public class MyBasicChannel {
    
    var channel:FlutterBasicMessageChannel
    
    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterBasicMessageChannel(name: "com.flutter.test.basic_channel", binaryMessenger: messenger, codec: FlutterStringCodec.sharedInstance())
        
        // Receive messages from Dart and send replies.
        channel.setMessageHandler {
          (message: Any?, reply: FlutterReply) -> Void in
            self.channel.sendMessage("from ios")
            print(message as! String)
        }

    }
}
